src/utils/user-list-helpers.js

```diff
@@ -0,0 +1,259 @@
+import {
+  getSolidDataset,
+  createThing,
+  buildThing,
+  setThing,
+  createSolidDataset,
+  getThingAll,
+  saveSolidDatasetAt,
+  getStringNoLocale,
+  getUrl,
+  getDatetime,
+  removeThing,
+  getPodUrlAll
+} from '@inrupt/solid-client';
+
+import { setDocAclForUser } from './network/session-helper';
+
+import { RDF_PREDICATES } from '../constants';
+
+/**
+ * @typedef {import('@inrupt/solid-ui-react').SessionContext} Session
+ */
+
+/**
+ * @typedef {import("../typedefs").userListObject} userListObject
+ */
+
+const createUsersList = async (session, usersListUrl) => {
+  const newUsersList = buildThing(createThing({ name: 'userlist' }))
+    .addStringNoLocale(RDF_PREDICATES.name, 'Users List')
+    .addStringNoLocale(RDF_PREDICATES.description, 'A list of users')
+    .build();
+
+  let newSolidDataset = createSolidDataset();
+  newSolidDataset = setThing(newSolidDataset, newUsersList);
+
+  // Generate document.ttl file for container
+  const usersListDataset = await saveSolidDatasetAt(usersListUrl, newSolidDataset, {
+    contentType: 'text/turtle',
+    fetch: session.fetch
+  });
+
+  // Generate ACL file for container
+  await setDocAclForUser(session, usersListUrl);
+  return usersListDataset;
+};
+
+const parseUserObjectFromThing = (userThing) => {
+  const person = getStringNoLocale(userThing, RDF_PREDICATES.Person);
+  const givenName = getStringNoLocale(userThing, RDF_PREDICATES.givenName);
+  const familyName = getStringNoLocale(userThing, RDF_PREDICATES.familyName);
+  const webId = getUrl(userThing, RDF_PREDICATES.identifier);
+  const podUrl = getUrl(userThing, RDF_PREDICATES.URL);
```

Comment:
Add Web ID and Pod Url separately here. Solid allows a user to have a web id at one URL, and a pod at a totally different url.